jenkins-scripts/dsl/_configs_/OSRFUNIXBase.groovy

```diff
@@ -38,6 +38,62 @@ class OSRFUNIXBase extends OSRFBase
              git clone https://github.com/gazebo-tooling/release-tools scripts -b \$RTOOLS_BRANCH
              """.stripIndent())
       }
+          publishers {
+            postBuildScripts {
+              steps{
+              conditionalSteps {
+                  condition {
+                      expression('(.)*gpu-nvidia(.)*','${NODE_LABELS}')
+                  }
+                  steps {
+                     systemGroovyCommand('''\
+                        import hudson.model.Cause.UpstreamCause;
+                        import hudson.model.*;
+                    
+                        def node = build.getBuiltOn()
+                        def old_labels = node.getLabelString()
+
+                        println("# BEGIN SECTION: NVIDIA MISMATCH RECOVERY")
+                        if (!(build.getLog(1000) =~ "nvml error: driver/library version mismatch")) {
+                          println("Build failed for other reason than NVIDIA error - Not performing any more steps")
+                          return 0;
+                        } else {
+                          try {
+                            println("Removing labels and adding 'recovery-process' label to node")
```

Comment:
Here we have detected the nvidia problem. May be can add a message before this one saying something like:
```groovy
println(" PROBLEM: NVIDIA driver/library version mismatch was detected in the log. Try to automatically resolve it:")
```